torch/_functorch/aot_autograd.py

```diff
@@ -2982,6 +2986,17 @@ def backward(ctx, *flat_args):
             del contiguous_args
 
             def call_compiled_backward():
+                if ctx._compiled_autograd_tracing:
+                    # For compiled autograd, run raw FX graph so that it can be inlined into the larger graph
+                    symints = ctx._compiled_autograd_symints
+                    assert len(symints) == len(ctx.symints)
+                    all_args[:len(symints)] = symints
```

Comment:
This is the thing that avoids allocating symints unless we've seen them vary, is that right?

Reply:
No, for AotAutograd's SymInts we assume everything is dynamic by default, because for static models `len(symints)==0`.  The lazy-symint stuff is done inside the cache node.